networkx/algorithms/tests/test_lowest_common_ancestors.py

```diff
@@ -425,3 +425,34 @@ def test_all_pairs_lca_self_ancestors():
     assert all(u == v == a for (u, v), a in ap_lca(MG) if u == v)
     MG.add_edges_from([(1, 0), (2, 0)])
     assert all(u == v == a for (u, v), a in ap_lca(MG) if u == v)
+
+
+def test_lca_on_null_graph():
+    G = nx.null_graph(create_using=nx.DiGraph)
+    with pytest.raises(
+        nx.NetworkXPointlessConcept, match="LCA meaningless on null graphs"
+    ):
+        nx.lowest_common_ancestor(G, 0, 0)
+
+
+def test_lca_on_cycle_graph():
+    G = nx.cycle_graph(6, create_using=nx.DiGraph)
+    with pytest.raises(
+        nx.NetworkXError, match="LCA only defined on directed acyclic graphs"
+    ):
+        nx.lowest_common_ancestor(G, 0, 3)
+
+
+def test_lca_multiple_valid_solutions():
+    G = nx.DiGraph()
+    G.add_nodes_from(range(4))
+    G.add_edges_from([(2, 0), (3, 0), (2, 1), (3, 1)])
+    assert nx.lowest_common_ancestor(G, 0, 1) in {2, 3}
+
+
+def test_lca_dont_rely_on_single_successor():
+    # Nodes 0 and 1 have nodes 2 and 3 as successors, and node 3 is a successor to 2.
```

Comment:
I'm still confused by the comment - I think it makes sense if "successor" is replaced with "ancestor", or if the nodes here are being referred to by index rather than value (i.e. `{0: 3, 1: 2, ... etc.}`, which would still be confusing!)

Just commenting in case I'm missing something! Otherwise, the executable parts of these changes LGTM!

Reply:
Another good catch! I've been known to be dyslexic around this and seem to get it wrong more than half the time. I updated the language to use "immediate ancestor", and I think it's clearer.